### PR TITLE
sec: conform to Gosec [G114](https://github.com/securego/gosec\#avail…

### PR DESCRIPTION
**Pull request by sober-wang**
- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

#### What type of this PR
Examples of user facing changes:
- sec: this cli listen http server conform to Gosec G114

#### Which issue(s) this PR fixes:
Fixes (#1859 )

### CODE REVIEW

File: cmd/controller/controller.go [hunk 1]
```diff
@@ -49,7 +49,14 @@ func CmdMain() {
...
 			mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
 		}
-		klog.Fatal(http.ListenAndServe(fmt.Sprintf("0.0.0.0:%d", config.PprofPort), mux))
+
+		// conform to Gosec G114
+		// https://github.com/securego/gosec#available-rules
+		server := &http.Server{
+			Addr:              fmt.Sprintf("0.0.0.0:%d", config.PprofPort),
+			ReadHeaderTimeout: 3 * time.Second,
+		}
```
Comment (zhangzujian) [inline cmd/controller/controller.go:58]: Missing handler here. `mux` need to be added to `server`.

File: cmd/daemon/cniserver.go [hunk 2]
```diff
@@ -94,7 +94,14 @@ func CmdMain() {
...
 		mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
 	}
-	klog.Fatal(http.ListenAndServe(fmt.Sprintf("0.0.0.0:%d", config.PprofPort), mux))
+
+	// conform to Gosec G114
+	// https://github.com/securego/gosec#available-rules
+	server := &http.Server{
+		Addr:              fmt.Sprintf("0.0.0.0:%d", config.PprofPort),
+		ReadHeaderTimeout: 3 * time.Second,
+	}
```
Comment (zhangzujian) [inline cmd/daemon/cniserver.go:103]: Missing handler, too.

**Comment by zhangzujian** [comment]
The version of `gosec` used by GitHub actions needs to be updated:

https://github.com/kubeovn/kube-ovn/blob/master/.github/workflows/build-windows.yaml#L20
https://github.com/kubeovn/kube-ovn/blob/master/.github/workflows/build-x86-image.yaml#L20

**Comment by sober-wang** [comment]
> The version of `gosec` used by GitHub actions needs to be updated:
> 
> https://github.com/kubeovn/kube-ovn/blob/master/.github/workflows/build-windows.yaml#L20 https://github.com/kubeovn/kube-ovn/blob/master/.github/workflows/build-x86-image.yaml#L20

yeh ,my `gosec` version is v2.12.0 , I'm sure.

图片

**Pull request merged**